workitem/link/link_repository.go

```diff
@@ -88,16 +96,28 @@ func (r *GormWorkItemLinkRepository) ValidateCorrectSourceAndTargetType(ctx cont
 		return errs.WithStack(err)
 	}
 	// Check type paths
```

Comment:
BTW, if this is the case, did you add tests to verify that this validation method behaves properly in all cases, including with valid and invalid subtypes ?